tests/controllers/cleanup/delete_tokens_test.go

```diff
@@ -0,0 +1,165 @@
+
+	// Create the wrangler and management contexts
+	wranglerContext, err := wrangler.NewContext(s.ctx, nil, restCfg)
+	assert.NoError(s.T(), err)
+
+	scaledContext, clusterManager, _, err := multiclustermanager.BuildScaledContext(s.ctx, wranglerContext, &multiclustermanager.Options{})
+	assert.NoError(s.T(), err)
+	s.managementContext, err = scaledContext.NewManagementContext()
+	assert.NoError(s.T(), err)
+
+	// Register the auth controller
+	auth.RegisterEarly(s.ctx, s.managementContext, clusterManager)
+
+	// Start controllers
+	common.StartNormanControllers(s.ctx, s.T(), s.managementContext,
+		schema.GroupVersionKind{
+			Group:   "management.cattle.io",
+			Version: "v3",
+			Kind:    "AuthConfig",
+		},
+	)
+
+	// Start caches
+	common.StartWranglerCaches(s.ctx, s.T(), s.managementContext.Wrangler,
+		schema.GroupVersionKind{
+			Group:   "management.cattle.io",
+			Version: "v3",
+			Kind:    "Token",
+		},
+	)
+}
+
+func (s *AuthConfigSuite) TestTokensCleanup() {
+	t := assert.New(s.T())
+
+	const (
+		tick         = 1 * time.Second
+		duration     = 10 * time.Second
+		numOfTokens  = 1000
+		authProvider = "openldap"
+	)
+
+	// create the AuthConfig for the authProvider
+	_, err := s.managementContext.Management.AuthConfigs("").Create(&v3.AuthConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: authProvider,
+		},
+		Type:    client.OpenLdapConfigType,
+		Enabled: true,
+	})
+	t.NoError(err)
+
+	// create some tokens for the authProvider
+	for i := 0; i < numOfTokens; i++ {
+		_, err := s.managementContext.Management.Tokens("").Create(&v3.Token{
+			ObjectMeta:   metav1.ObjectMeta{GenerateName: "t-"},
+			AuthProvider: authProvider,
+		})
+		t.NoError(err)
+	}
+
+	// and also some local tokens
+	for i := 0; i < numOfTokens; i++ {
+		_, err := s.managementContext.Management.Tokens("").Create(&v3.Token{
+			ObjectMeta:   metav1.ObjectMeta{GenerateName: "t-"},
+			AuthProvider: "local",
+		})
+		t.NoError(err)
+	}
+
+	fmt.Printf("created %d tokens\n", numOfTokens)
+
+	// disable the provider, setting the cleanup annotation (eventually because of retries)
+	authConfig, err := s.managementContext.Management.AuthConfigs("").Get(authProvider, metav1.GetOptions{})
+	t.NoError(err)
+
+	authConfig.Enabled = false
+	authConfig.Annotations = map[string]string{auth.CleanupAnnotation: auth.CleanupUnlocked}
+	authConfig, err = s.managementContext.Management.AuthConfigs("").Update(authConfig)
+	t.NoError(err)
+
+	// check that all the authProvider tokens are deleted
+	ok := assert.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+		tokens, err := s.managementContext.Management.Tokens("").List(metav1.ListOptions{})
+		assert.NoError(c, err)
+		assert.NotNil(c, tokens)
+
+		var remainingTokens int
+		for _, token := range tokens.Items {
+			if token.AuthProvider == authProvider {
+				remainingTokens++
+			}
+		}
+
+		fmt.Printf("remaining %d tokens\n", remainingTokens)
+		assert.Zero(c, remainingTokens)
+	}, duration, tick)
+
+	t.True(ok, "not ok")
```

Comment:
We can maybe come up with a better message here?

"secrets were left over after disabling auth configuration" or something like that?